### PR TITLE
Refactor openshift node role

### PR DESCRIPTION
**Pull request by ingvagabund**
* move disable swap tasks into `disable_swap.yml`
* introduce `r_openshift_node_` prefixed variables and remove all free variables
* autogenerate table of role variables
* move node installation into its own file

Built on top of https://github.com/openshift/openshift-ansible/pull/5025. Once the underlying PR gets merged, I will rebase.

**In depth**
In a process of moving all global variables out of the `openshift_node` role I had to do the following:
1. For each global variable used inside the role create corresponding new variable prefixed with `r_openshift_node_`  string, e.g. for `openshift_node_sdn_mtu` create `r_openshift_node_facts_sdn_mtu`.
2. Given most of the variables has default value attach, I moved the default value of each role variable under `defaults/main.yml`. E.g. the `openshift.node.env_vars | default({})` resulted in `r_openshift_node_env_vars: {}` line.
3. Set all role variables to their corresponding global variables. By the nature of `openshift.node.env_vars | default({})` expression, one can use the following form:
   ```yaml
   include_role:
     name: openshift_node
   vars:
     r_openshift_node_env_vars: "{{ openshift.node.env_vars | default({}) }}"
   ```

   However, the default value (that is set inside `default/main.yml` gets duplicated). For that reason I decided to use the following form instead:

   ```yaml
   set_fact:
      r_openshift_node_env_vars: "{{ openshift.node.env_vars }}"
   when: openshift.node.env_vars is defined

   include_role:
     name: openshift_node
   ```
   It does not look good but the default value is specified only once. Plus, the `set_fact: ... when: ...` syntax is pretty generic, it can be automatically generated (see `playbooks/common/openshift-node/set_optional_node_facts_role_variables.yml`)
4. Verify role variables: a role has a list of role variables that needs to be defined, a list of variables that can be set to a restricted set of value or a list of variables that depends on other variables or their values. 
The checks can be automatically generated (see roles/openshift_node/tasks/pre_checks.yml).

As you can see in this PR there is about 50 role variables that are being set before the `openshift_node` role gets invoked. The fact may be a signal the role needs to be broken out into smaller roles.

### CODE REVIEW

**Comment by ingvagabund** [comment]
aos-ci-test

**Comment by ingvagabund** [comment]
aos-ci-test

**Comment by ingvagabund** [comment]
aos-ci-test

**Comment by ingvagabund** [comment]
aos-ci-test

File: playbooks/common/openshift-node/config.yml [hunk 1]
```diff
@@ -46,6 +47,8 @@
                                                     | union(groups['oo_etcd_to_config'] | default([])))
                                                 | oo_collect('openshift.common.hostname') | default([]) | join (',')
                                                 }}"
+  pre_tasks:
+  - include: pre_install_node.yml
   roles:
```
Comment (ingvagabund) [inline playbooks/common/openshift-node/config.yml]: If I use
```yaml
- include_role:
    name: openshift_node
  vars:
    openshift_ca_host: "{{ groups.oo_first_master.0 }}"
```
instead, the `restart NetworkManager` and `restart dnsmasq` handlers get skipped:

```
RUNNING HANDLER [openshift_node_dnsmasq : restart NetworkManager] **************
skipping: [10.8.170.180] => {
    "changed": false, 
    "skip_reason": "Conditional result was False", 
    "skipped": true
}

RUNNING HANDLER [openshift_node_dnsmasq : restart dnsmasq] *********************
skipping: [10.8.170.180] => {
    "changed": false, 
    "skip_reason": "Conditional result was False", 
    "skipped": true
}
META: ran handlers
```
even though there is no `when: ` condition attached to any of the handlers.

**Comment by ingvagabund** [comment]
@kwoodson once this PR is merged, I will continue with breaking the `openshift_node` role into smaller roles, e.g.
- `openshift_node`
- `openshift_node_storage`
- `openshift_node_networking`
- ...

Or, at least make a survey how realistic that is.

**Comment by ingvagabund** [comment]
aos-ci-test

**Comment by ingvagabund** [comment]
aos-ci-test

**Comment by ingvagabund** [comment]
aos-ci-test

**Comment by ingvagabund** [comment]
[test]

**Comment by michaelgugino** [comment]
Closing due to inactivity.

**Pull request closed without merging**